packages/jupyter-ai/jupyter_ai/personas/jupyternaut/jupyternaut.py

```diff
@@ -27,19 +27,24 @@ def defaults(self):
         )
 
     async def process_message(self, message: Message) -> None:
-        provider_name = self.config_manager.lm_provider.name
-        model_id = self.config_manager.lm_provider_params["model_id"]
-
-        runnable = self.build_runnable()
-        variables = JupyternautVariables(
-            input=message.body,
-            model_id=model_id,
-            provider_name=provider_name,
-            persona_name=self.name,
-        )
-        variables_dict = variables.model_dump()
-        reply_stream = runnable.astream(variables_dict)
-        await self.stream_message(reply_stream)
+        if self.config_manager.lm_provider:
+            provider_name = self.config_manager.lm_provider.name
+            model_id = self.config_manager.lm_provider_params["model_id"]
+
+            runnable = self.build_runnable()
+            variables = JupyternautVariables(
+                input=message.body,
+                model_id=model_id,
+                provider_name=provider_name,
+                persona_name=self.name,
+            )
+            variables_dict = variables.model_dump()
+            reply_stream = runnable.astream(variables_dict)
+            await self.stream_message(reply_stream)
+        else:
+            self.send_message(
+                "No language model provider configured. Please set one in the Jupyter AI settings."
+            )
```

Comment:
Instead having if-else statement you could put this error handling logic into an early return / guard clause under condition like `if not self.config_manager.lm_provider:` and then logic from `if` branch would not have to be nested.

This would reduce nesting levels and make code easier to read.